packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource.ts

```diff
@@ -133,6 +135,7 @@ export class BucketNotifications extends Construct {
           BucketName: this.bucket.bucketName,
           NotificationConfiguration: cdk.Lazy.any({ produce: () => this.renderNotificationConfiguration() }),
           Managed: managed,
+          S3NotificationsDeleteFeatureFlagEnabled: FeatureFlags.of(this).isEnabled(S3_EXISTING_NOTIFICATIONS_DELETE_ENABLED),
```

Comment:
I think this name is a bit misleading. We're not actually changing the `Delete` event but more of the `Update` event, and it's not something that users can modified so we don't need to mention it's a feature flag. Can we change the name to something like `ApplyNameTransformations` or anything better that you can think of.

Reply:
How about ApplyS3EventNotificationNameTransformations?

Reply:
IMO S3EventNotification is a bit redundant since this custom resource handler is for s3 notification already.